atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/DefaultTargetFactory.java

```diff
@@ -0,0 +1,139 @@
+            AuxiliaryRemotingParameters parameters) {
+        ClientOptions relevantOptions = ClientOptions.fromRemotingParameters(parameters);
+        ClientConfiguration clientConfiguration = relevantOptions.create(
+                ImmutableList.of(uri),
+                Optional.empty(),
+                trustContext.orElseThrow(() -> new IllegalStateException("CJR requires a trust context")));
+        return client(type, clientConfiguration);
+    }
+
+    @Override
+    public <T> T createProxyWithFailover(ServerListConfig serverListConfig, Class<T> type,
+            AuxiliaryRemotingParameters parameters) {
+        // It doesn't make sense to create a proxy with the capacity to failover that doesn't retry.
+        ClientOptions clientOptions = getClientOptionsForFailoverProxy(parameters);
+        return client(type, clientOptions.serverListToClient(serverListConfig));
+    }
+
+    @Override
+    public <T> T createLiveReloadingProxyWithFailover(Supplier<ServerListConfig> serverListConfigSupplier,
+            Class<T> type, AuxiliaryRemotingParameters parameters) {
+        ClientOptions options = getClientOptionsForFailoverProxy(parameters);
+        Supplier<T> clientSupplier = new CachedTransformingSupplier<>(
+                serverListConfigSupplier,
+                serverListConfig -> client(type, options.serverListToClient(serverListConfig)));
+        return FastFailoverProxy.newProxyInstance(type, clientSupplier);
+    }
+
+    public <T> T createProxyWithQuickFailoverForTesting(
+            ServerListConfig serverListConfig,
+            Class<T> type,
+            AuxiliaryRemotingParameters parameters) {
+        return client(type, ClientOptions.FAST_RETRYING_FOR_TEST.serverListToClient(serverListConfig));
+    }
+
+    private static ClientOptions getClientOptionsForFailoverProxy(AuxiliaryRemotingParameters parameters) {
+        return ClientOptions.fromRemotingParameters(
+                ImmutableAuxiliaryRemotingParameters.copyOf(parameters).withShouldRetry(true));
+    }
+
+    private static <T> T client(Class<T> clazz, ClientConfiguration clientConfiguration) {
+        if (isDialogue(clazz)) {
+            return dialogueClient(clazz, clientConfiguration);
+        }
+        return conjureClient(clazz, clientConfiguration);
+    }
+
+    private static <T> T conjureClient(Class<T> clazz, ClientConfiguration clientConfiguration) {
+        return JaxRsClient.create(
+                clazz,
+                clientConfiguration.userAgent()
+                        .orElseThrow(() -> new SafeIllegalStateException("User agent must be provided")),
+                HOST_METRICS_REGISTRY,
+                clientConfiguration);
+    }
+
+    private static <T> T dialogueClient(Class<T> clazz, ClientConfiguration clientConfiguration) {
+        return callStaticFactoryMethod(clazz, ApacheHttpClientChannels.create(clientConfiguration));
```

Comment:
Every call to `ApacheHttpClientChannels.create(clientConfiguration)` creates a new connection pool. I think when timelock is blue/greening it would be cool if we could use the `updateUris` method and avoid re-opening a ton of connections?

Reply:
baby steps Dan. Just trying to get it to run first